syntax = "proto3";

package process_service;

service ImageService {
  // 客户端流式上传图片
  rpc UploadImage(stream ImageChunk) returns (ImageResponse) {}
}

message ImageChunk {
  oneof data {
    // 第一个请求包含元数据
    ImageInfo info = 1;
    // 后续请求包含图片数据块
    bytes chunk = 2;
  }
}

message ImageInfo {
  string filename = 1;
  string image_type = 2; // 例如 "png", "jpg"
}

message ImageResponse {
  string text = 1; // 识别的文字
  string error = 2; // 错误信息
}